src/USER-GWBC/fix_wall_ACL.cpp

```diff
@@ -0,0 +1,284 @@
+/* ----------------------------------------------------------------------
+   LAMMPS - Large-scale Atomic/Molecular Massively Parallel Simulator
+   http://lammps.sandia.gov, Sandia National Laboratories
+   Steve Plimpton, sjplimp@sandia.gov
+
+   Copyright (2003) Sandia Corporation.  Under the terms of Contract
+   DE-AC04-94AL85000 with Sandia Corporation, the U.S. Government retains
+   certain rights in this software.  This software is distributed under
+   the GNU General Public License.
+
+   See the README file in the top-level LAMMPS directory.
+------------------------------------------------------------------------- */
+
+#include "fix_wall_ACL.h"
+#include <cstring>
+#include "atom.h"
+#include "comm.h"
+#include "update.h"
+#include "modify.h"
+#include "domain.h"
+#include "lattice.h"
+#include "input.h"
+#include "variable.h"
+#include "error.h"
+#include "force.h"
+#include "random_extra.h"
+#include <stdlib.h> 
+/*#include <iostream> */
+
+using namespace LAMMPS_NS;
+using namespace FixConst;
+/* using namespace std; */
+
+enum{XLO=0,XHI=1,YLO=2,YHI=3,ZLO=4,ZHI=5};
+enum{NONE=0,EDGE,CONSTANT,VARIABLE};
+
+/* ---------------------------------------------------------------------- */
+
+FixWallACL::FixWallACL(LAMMPS *lmp, int narg, char **arg) :

```

Comment:
since this fix (and the others) seem to share a lot of ode with fix wall/reflect, wouldn't it be better to have them all derived from FixWallReflect? processing of common command line arguments could then be handled in the base class and derived classes could then moved to and handled in an "options()"  method similar as in, e.g., the various region styles.